app/controllers/rentalsandrentaroom/allowances/RentalsRaRAllowancesCompleteController.scala

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.rentalsandrentaroom.allowances
+
+import controllers.ControllerUtils.statusForPage
+import controllers.actions._
+import forms.rentalsandrentaroom.allowances.RentalsRaRAllowancesCompleteFormProvider
+import models.JourneyContext
+import pages.rentalsandrentaroom.allowances.RentalsRaRAllowancesCompletePage
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import service.JourneyAnswersService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.rentalsandrentaroom.allowances.RentalsRaRAllowancesCompleteView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class RentalsRaRAllowancesCompleteController @Inject()(
```

Comment:
Could you make this controller more generic? I hope you recall the discussion we had during the refactoring session. Discuss with Costas or we can have a tech huddle tomorrow

Reply:
@cmantzoukashmrc